scrapy_poet/overrides.py

```diff
@@ -89,26 +91,27 @@ def from_crawler(cls, crawler: Crawler) -> Crawler:
         return cls(crawler.settings.getlist("SCRAPY_POET_OVERRIDES", []))
 
     def __init__(self, rules: Optional[Iterable[RuleFromUser]] = None) -> None:
-        self.rules: List[OverrideRule] = []
-        self.matcher: Dict[Callable, URLMatcher] = defaultdict(URLMatcher)
+        self.rules: List[ApplyRule] = []
+        self.matcher: Dict[PageObject, URLMatcher] = defaultdict(URLMatcher)
         for rule in rules or []:
             self.add_rule(rule)
-        logger.debug(f"List of parsed OverrideRules:\n{self.rules}")
+        logger.debug(f"List of parsed ApplyRules:\n{self.rules}")
 
     def add_rule(self, rule: RuleFromUser) -> None:
         if isinstance(rule, (tuple, list)):
             if len(rule) != 3:
                 raise ValueError(
-                    f"Invalid overrides rule: {rule}. Rules as tuples must have "
+                    f"Invalid rule: {rule}. Rules as tuples must have "
                     f"3 elements: (1) the pattern, (2) the PO class used as a "
                     f"replacement and (3) the PO class to be replaced."
                 )
             pattern, use, instead_of = rule
-            rule = OverrideRule(
+            rule = ApplyRule(
                 for_patterns=Patterns([pattern]), use=use, instead_of=instead_of
             )
         self.rules.append(rule)
-        self.matcher[rule.instead_of].add_or_update(
+        # FIXME: This key will change with the new rule.to_return
+        self.matcher[rule.instead_of].add_or_update(  # type: ignore
```

Comment:
Adding a comment here so it is not forgotten.

Reply:
This one is handled in one of the commits from the other PR: https://github.com/scrapinghub/scrapy-poet/pull/88/commits/a58b250eaef862ce31ce2a844b5970646b3b1822#diff-4906cda8a37b6f984ea889eab98b820f6b9f6908f7994ab26e50adfdeff95004L113